ArchiSteamFarm/IPC/Integration/ApiAuthenticationMiddleware.cs

```diff
@@ -87,18 +87,22 @@ internal sealed class ApiAuthenticationMiddleware {
 				throw new InvalidOperationException(nameof(ClearFailedAuthorizationsTimer));
 			}
 
-			string? ipcPassword = ASF.GlobalConfig != null ? ASF.GlobalConfig.IPCPassword : GlobalConfig.DefaultIPCPassword;
-
-			if (string.IsNullOrEmpty(ipcPassword)) {
-				return HttpStatusCode.OK;
-			}
-
 			IPAddress? clientIP = context.Connection.RemoteIpAddress;
 
 			if (clientIP == null) {
 				throw new InvalidOperationException(nameof(clientIP));
 			}
 
+			string? ipcPassword = ASF.GlobalConfig != null ? ASF.GlobalConfig.IPCPassword : GlobalConfig.DefaultIPCPassword;
```

Comment:
I think we could do with some syntactic sugar here.
```suggestion
			string? ipcPassword = ASF.GlobalConfig?.IPCPassword ?? GlobalConfig.DefaultIPCPassword;
```

Reply:
And this is where you're **wrong** 😆. We want to fallback only if there is no global config, **not** when there is no global config **or** no ipc password specified.

Just because current behaviour of default `IPCPassword` is to be null, doesn't mean it'll always stay like that. If you did the same with `SteamMessagePrefix` for example (which we have in other place), you'd falsely default to `/me ` instead of `` when the user set it to `null` himself.

Nope.

Reply:
True that. I didn't think of this edge case.

Reply:
Me neither, I found it as a regression when doing this exact thing during my nullable cleanup round, and realized this doesn't do what was there previously.